pkg/asset/installconfig/azure/session.go

```diff
@@ -130,11 +115,41 @@ func GetSessionWithCredentials(cloudName azure.CloudEnvironment, armEndpoint str
 	if err != nil {
 		return nil, err
 	}
-	session.CloudConfig = cloudConfig
+	session.CloudConfig = *cloudConfig
 	session.AuthType = authType
 	return session, nil
 }
 
+// GetCloudConfiguration gets a cloud configuration from the cloud name and endpoint.
+func GetCloudConfiguration(cloudName azure.CloudEnvironment, armEndpoint string) (*cloud.Configuration, error) {
+	cloudEnv, err := azureenv.EnvironmentFromURL(armEndpoint)
```

Comment:
This is missing the non-stackcloud case https://github.com/openshift/installer/pull/8669/files#diff-da950a4ddabbede621d9d3b1058bb34f8931c89179306ee88a0e4d76a4cf0b13R74-R81.

This is causing the failure in https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_installer/8669/pull-ci-openshift-installer-master-e2e-azure-ovn/1811162760466141184